cmd/common-main.go

```diff
@@ -50,24 +50,42 @@ func initConfig() {
 	}
 }
 
-func handleCommonCmdArgs(ctx *cli.Context) {
-	// Set configuration directory.
-	{
-		// Get configuration directory from command line argument.
-		configDir := ctx.String("config-dir")
-		if !ctx.IsSet("config-dir") && ctx.GlobalIsSet("config-dir") {
-			configDir = ctx.GlobalString("config-dir")
+func handleGlobalCmdArgs(ctx *cli.Context) error {
+	if !globalJSON {
+		// Get "json" flag from command line argument and
+		// enable json and quiet modes if jason flag is turned on.
+		globalJSON = ctx.Bool("json") || ctx.GlobalBool("json")
+		if globalJSON {
+			log.EnableJSON()
+		}
+	}
+
+	if !globalQuiet {
+
+		// Get quiet flag from command line argument.
+		globalQuiet = ctx.Bool("quiet") || ctx.GlobalBool("quiet")
+		if globalQuiet {
+			log.EnableQuiet()
 		}
-		if configDir == "" {
-			fatalIf(errors.New("empty directory"), "Configuration directory cannot be empty.")
+	}
+	if globalConfigDir == "" {
```

Comment:
On what case you have non-empty value before command line arg parsing?

Reply:
Wanted to reuse the function. It is used twice in gateway-main.go.

Reply:
Couldn't able to understand.  This function is called twice in gateway-main?  Isn't it a bug in gateway-main?